cmd/server-main.go

```diff
@@ -97,12 +98,57 @@ EXAMPLES:
 
 type serverCmdConfig struct {
 	serverAddr   string
-	disks        []string
-	ignoredDisks []string
+	disks        []storageEndPoint
+	ignoredDisks []storageEndPoint
 	isDistXL     bool // True only if its distributed XL.
 	storageDisks []StorageAPI
 }
 
+type storageEndPoint struct {
+	host string
+	port int
+	path string
+}
+
+func (ep storageEndPoint) String() string {
+	var str []string
+	if ep.host != "" {
+		str = append(str, ep.host)
+	}
+	if ep.port != 0 {
+		str = append(str, strconv.Itoa(ep.port))
+	}
+	if ep.path != "" {
+		str = append(str, ep.path)
+	}
+	return strings.Join(str, ":")
+}
+
+func parseStorageEndPoint(ep string, defaultPort int) storageEndPoint {
+	parts := strings.Split(ep, ":")
```

Comment:
strings.Split() will be better because `127.0.0.1:80:/disk1:/disk1` will result in parse error in the default case of the switch statement here, if SplitN is used it will not catch it.